front/lib/api/assistant/agent.ts

```diff
@@ -583,8 +567,23 @@ export async function* runMultiActionsAgent(
   }
 
   const actions: AgentActionsEvent["actions"] = [];
+  const agentActions = agentConfiguration.actions;
+
+  if (agentActions.length === 1 && !agentActions[0].name) {
+    // Special case for legacy single-action agents that have never been edited in
+    // multi-actions mode.
+    // We must backfill the name from the legacy spec in order to match the action.
```

Comment:
I was wondering: why not running a migration to fill all the empty action name/desc from the spec? So that fields are mandatory in db and we don't need all the fallbacks.

Reply:
I don't want people to see those fake descriptions in their builder. For the name we def could I agree

Reply:
Especially for retrieval / tables query, the description people put is not the full description of the action so we have no way to backfill something that makes sense

Reply:
Yes but seeing some weird description might makes them fix it, as opposed to hiding the fact we will use a fake description if they don't provide one!

Reply:
No but it will simply create super incoherent prompts

Reply:
Because then we won't know that we're still in "legacy mode", so we'll put that automatic description through this thing:
https://github.com/dust-tt/dust/blob/0de2b661d63fb70829a96a39a797468f78568f4e/front/lib/api/assistant/actions/retrieval.ts#L264-L287

Reply:
Whe we really want to use the legacy description (from legacy spec)